### PR TITLE
CLI uses ballista context instead of datafusion context in local mode

### LINKED ISSUE #219: CLI uses DataFusion context if no host or port are provided
**Describe the bug**
The Ballista CLI executes queries against a DataFusion context, not Ballista, if no host and port are provided.

**To Reproduce**
:point_up: 

**Expected behavior**
We should either remove this option or make it use Ballista in local mode.

**Additional context**
None

### PR DESCRIPTION
**Pull request by r4ntix**
# Which issue does this PR close?

Closes #219

 # Rationale for this change

See #219 

# What changes are included in this PR?

- add concurrent_tasks parameter for ballista local mode, default to all available cores
- use `BallistaContext` instead of `SessionContext`
- use `BallistaError` instead of `DataFusionError`

# Are there any user-facing changes?

add concurrent_tasks parameter for ballista local mode, default to all available cores

### CODE REVIEW

File: ballista-cli/src/context.rs [hunk 1]
```diff
@@ -17,56 +17,52 @@
...
-            Context::Local(datafusion) => datafusion.sql(sql).await,
-            Context::Remote(ballista) => ballista.sql(sql).await,
+            Context::Local(ballista) => ballista
+                .sql(sql)
+                .await
+                .map_err(BallistaError::DataFusionError),
+            Context::Remote(ballista) => ballista
+                .sql(sql)
+                .await
+                .map_err(BallistaError::DataFusionError),
```
Comment (andygrove) [inline ballista-cli/src/context.rs:65]: We have the same code in both cases. I think we can remove the `Context` abstraction in the CLI now and just use the Ballista context directly.

**Comment by andygrove** [comment]
This is looking good. Thanks @r4ntix. I left some feedback.

File: ballista-cli/src/context.rs (hunk 1 above)
Comment (r4ntix) [reply ballista-cli/src/context.rs:65]: 👍  Thanks for the feedback, I will fix it again.

Comment (r4ntix) [reply ballista-cli/src/context.rs]: I submit a new commit for this.

**Comment by andygrove** [comment]
@r4ntix I took the liberty of merging latest from master and fixing merge conflicts

**Review by andygrove — APPROVED** [review]

**Comment by r4ntix** [comment]
> @r4ntix I took the liberty of merging latest from master and fixing merge conflicts

Thank you very much for helping to fix the remaining issues, I will keep learning and contributing to ballista :)

**Pull request merged**